core/src/main/scala/scalaz/ReaderWriterStateT.scala

```diff
@@ -3,7 +3,10 @@ package scalaz
 /** A monad transformer stack yielding `(R, S1) => F[(W, A, S2)]`. */
 sealed abstract class IndexedReaderWriterStateT[F[_], -R, W, -S1, S2, A] {
   self =>
-  def run(r: R, s: S1): F[(W, A, S2)]
+  
+  def getF[S <: S1, RR <: R]: Monad[F] => F[(RR, S) => F[(W, A, S2)]]
+  
+  def run(r: R, s: S1)(implicit F: Monad[F]): F[(W, A, S2)] = F.join( F.map[(R, S1) => F[(W, A, S2)], F[(W, A, S2)]](getF(F))(f => f(r, s)) )
```

Comment:
minor point, but please put sig on first line and impl on second. It is quite hard to see that this is now not abstract.